### PR TITLE
MGMT-8999: Add hosts tab is displayed for an SNO cluster created via OCM

### PR DESCRIPTION
**Pull request by jgyselov**
https://issues.redhat.com/browse/MGMT-8999

### CODE REVIEW

File: src/ocm/components/AddHosts/utils.ts [hunk 1]
```diff
@@ -1,27 +1,22 @@
...
     ) {
-      isAllowed = true;
-    }
-
-    if (cluster.aiCluster && cluster.aiCluster['high_availability_mode'] === 'None') {
-      isAllowed = false;
+      return true;
+    } else if (
+      (cluster.state === 'ready' && cluster.product?.id == 'OCP-AssistedInstaller') ||
+      cluster.cloud_provider?.id == 'baremetal'
```
Comment (jkilzi) [inline src/ocm/components/AddHosts/utils.ts:15]:
```suggestion
      cluster.cloud_provider?.id === 'baremetal'
```

Comment (jgyselov) [reply src/ocm/components/AddHosts/utils.ts]: thx

File: src/ocm/components/AddHosts/utils.ts [hunk 2]
```diff
@@ -1,27 +1,22 @@
...
     ) {
-      isAllowed = true;
-    }
-
-    if (cluster.aiCluster && cluster.aiCluster['high_availability_mode'] === 'None') {
-      isAllowed = false;
+      return true;
+    } else if (
+      (cluster.state === 'ready' && cluster.product?.id == 'OCP-AssistedInstaller') ||
+      cluster.cloud_provider?.id === 'baremetal'
```
Comment (jkilzi) [inline src/ocm/components/AddHosts/utils.ts:15]: cluster.aiCluster might be defined && the entire condition holds && isSNO(cluster.aiCluster) === true -> BOOM!

Comment (jkilzi) [reply src/ocm/components/AddHosts/utils.ts:15]: I think we should return false immediately if it is SNO, regardless of the status

Comment (jgyselov) [reply src/ocm/components/AddHosts/utils.ts:15]: So like switch the order? First we check if `cluster.aiCluster` is defined, then if it's not SNO and the status last?  @jkilzi

Comment (jkilzi) [reply src/ocm/components/AddHosts/utils.ts:15]:
```javascript
if (cluster.aiCluster) {
      if (!isSNO(cluster.aiCluster)) {
        return cluster.aiCluster.status === "installed";
      } else {
        return false;
      }
    }
```

Comment (jgyselov) [reply src/ocm/components/AddHosts/utils.ts:15]: LGTM, if we want to nest the conditions. :shrug:

Comment (jgyselov) [reply src/ocm/components/AddHosts/utils.ts:15]: The logic stays the same -- all three conditions must be true for the function to be true.

File: src/ocm/components/AddHosts/types.ts [hunk 3]
```diff
@@ -37,5 +37,16 @@ export type OcmClusterType = {
...
 
+  metrics?: {
+    nodes: [
+      {
+        compute?: number;
+        infra?: number;
+        master?: number;
+        total?: number;
+      },
+    ];
```
Comment (jkilzi) [inline src/ocm/components/AddHosts/types.ts:48]: OMG, this is not always an array!!!

image

**Review by jkilzi — APPROVED** [review]

**Pull request merged**